### PR TITLE
feat: Refactor authentication provider and add documentation

### PR DESCRIPTION
**Pull request by joonatan-aatos**
## WHY:

Authentication code required refactoring.

### What has been changed (if possible, add screenshots, gifs, etc. )

The structure of the code was changed.
Documentation comments were added.

## Check off each of the following tasks as they are completed

- [x] I have reviewed the changes myself in this PR. Please check the [self-review document](https://github.com/OpenVAA/voting-advice-application/blob/main/docs/contributing/self-review.md)
- [ ] I have added or edited unit tests.
- [x] I have run the unit tests successfully.
- [x] I have tested this change on my own device.
- [ ] I have tested this change on other devices (Using Browserstack is recommended).
- [x] I have tested my changes using the [WAVE extension](https://wave.webaim.org/extension/)
- [x] I have added documentation where necessary.
- [ ] Is there an existing issue linked to this PR?

**Clean up your git commit history before submitting the pull request!**

### CODE REVIEW

File: frontend/src/lib/components/navigation/Navigation.svelte [hunk 1]
```diff
@@ -3,7 +3,8 @@
   import {candidateAppRoute} from '$candidate/placeholder.json';
   import TimedModal from './TimedModal.svelte';
   import {goto} from '$app/navigation';
-  import {authContext} from '../authentication/authenticationStore';
+  import {authContext} from '$lib/utils/authenticationStore';
+  import RequireLogin from '../authentication/RequireLogin.svelte';
```
Comment (TrixterTheTux) [inline frontend/src/lib/components/navigation/Navigation.svelte:7]: `$lib/components/authentication/RequireLogin.svelte`

File: frontend/src/lib/components/navigation/Navigation.svelte [hunk 2]
```diff
@@ -41,282 +42,286 @@
...
+<RequireLogin>
+  <div class="drawer">
+    <TimedModal
+      bind:timeLeftInt
+      bind:openModal
+      bind:closeModal
+      onTimeout={logout}
+      timerDuration={logoutModalTimer}>
+      <div class="notification text-center text-black">
+        <h1>Some Of Your Data Is Still Missing</h1>
```
Comment (TrixterTheTux) [inline frontend/src/lib/components/navigation/Navigation.svelte:55]: Not directly caused by this PR, but missing localization

File: frontend/src/lib/components/navigation/Navigation.svelte [hunk 3]
```diff
@@ -41,282 +42,286 @@
...
+        <br />
+        <p>
+          There are still 10 items of basic info and 25 opinions to fill. Your data won’t be shown
+          in the Election Compass until you have filled these, but you can login later to continue.
```
Comment (TrixterTheTux) [inline frontend/src/lib/components/navigation/Navigation.svelte:59]: Not directly caused by this PR, but missing localization

File: frontend/src/lib/components/navigation/Navigation.svelte [hunk 4]
```diff
@@ -41,282 +42,286 @@
...
+        </p>
+        <p>
+          Are you sure you want to logout? You will be automatically logged out after {timeLeftInt}
+          seconds.
```
Comment (TrixterTheTux) [inline frontend/src/lib/components/navigation/Navigation.svelte:63]: Not directly caused by this PR, but missing localization

**Review by TrixterTheTux — CHANGES_REQUESTED** [review]

**Review by TrixterTheTux — APPROVED** [review]
Minor nitpick: `<!-- Move logout modal logic into its own component -->` could be prefixed with TODO: so it can be easily tracked by IDEs/code search.

**Review by norppaa — APPROVED** [review]

**Pull request merged**